README
$Id: README 1785 2006-05-18 02:06:18Z comdog $

You can install this using in the usual Perl fashion

	perl Makefile.PL
	make
	make test
	make install
	
The documentation is in the module file.  Once you install
the file, you can read it with perldoc.

	perldoc Module::Release

Look at the example script in script/release.
	
This module is also in CVS on SourceForge

	http://sourceforge.net/projects/brian-d-foy/


Enjoy, 

brian d foy, bdfoy@cpan.org

Changes
$Id: Changes 2674 2008-08-16 06:50:07Z comdog $

2.00_03 - Fri Aug 15 16:45:23 2008
	* Distro cleanup, no big whoop

2.00_02 - Sat Aug  9 03:41:07 2008
	* The beginning of the new Module::Release. Lots of stuff 
	moved into separate modules to act as mixins. There is a
	lot more testing. 
	* This is still a developer release, so don't get too
	attached to anything just yet.


1.22 - Sun Apr  6 13:13:30 2008
	* BUG FIX: Move kwalitee testing after normal testing, and have
	the kwalitee tests make their own dist to ensure that they
	have it.

1.21 - Sat Apr  5 14:54:20 2008
	* NEW FEATURE: Now test with multiple perls!

1.20 - Tue Nov 20 07:45:51 2007
	* fix dist_version to handle dev versions. Previously it was
	ignoring the _\d+, so the version that showed up in changes
	didn't have the _\d+.

1.19 - Sun Nov 11 04:10:22 2007
	* Sagar Shah noticed I tried to call _die() before I used bless().
	Oops. Fixed.
	* If you have older versions before I started using _die(), you don't
	need to rush to get this version. If you got the latest to use 
	Module::Release::Git, you need to upgrade.

1.18 - Sat Oct 27 22:19:57 2007
	* distro cleanups after moving from CVS to SVN
	* Changes from Torsten Schoenfeld to make get_readme and get_changes
	subclassable (that is, call them as methods like everything else)
	* Require Perl 5.006. I haven't done anything intentional to not
	use Perl 5.005, but I don't want to support that. With CPAN Testers
	going back farther in time now, I have to draw the line somewhere
	just to manage my time well. :)

1.17_01 - Wed Oct  3 21:52:02 2007
	* The output and debugging filehandles come from methods now. 
	This makes it easier to move the output around, or even turn
	it off. I wanted this for Module::Release::Git. The interface
	has some new features, but should be compatible with the older
	stuff.
	* Added some git support to release(1). This should refactor 
	itself out in various ways, but I just need the support right 
	now without going through all the work to make it nice nice.

1.17 - Mon Aug 13 13:43:17 2007
	* distro cleanups - no need to upgrade

1.16 - Thu May  3 10:08:14 2007
	* Fix for test suite problem

1.14 - Fri Mar 16 18:51:26 2007
	* use a temp file for cookies
	* don't test script if M::R::Subversion isn't installed

1.13 - Thu Feb 22 10:16:18 2007
	* Updated the distro for PAUSE indexing problems (my fault)
	* release(1) now detects SVN and loads Module::Release::Subversion 
	to handle it (that's a separate distro, though).
	* release(1) now uses Module::CPANTS::Analyze's cpants_lint.pl
	to check kwalitee. That's also a separate distro.

1.12 - Mon Sep 25 22:38:35 CDT 2006

	* Changes from Ken Williams
	
	* New 'makefile_PL' and 'makefile' parameters to the new()
	method allow Module::Build-based distributions to work more
	smoothly - set them to 'Build.PL' and 'Build', respectively.
	
	* The 'release_subclass' parameter to new() is un-deprecated
	again.
	
	* Subclasses previously could not override the new() method
	when invoked via the 'release_subclass' mechanism, now they can.
	
	* When running under a passive FTP mode, we were telling the
	server to go into passive mode twice, which causes some
	servers to bomb out.  This is fixed.
	
	* The error message when an FTP PUT operation failed has been
	improved.
	
	* System calls made during the release process will now read
	their underlying STDOUT and STDERR using read(), which helps
	you see what's going on when there are interactive steps.
	
	* The getpass() method will now check for CPAN_PASS and
	SF_PASS values previously set as member variables.  This
	helps users who might want to, for example, look their
	values up in the OS X keychain using a custom subclass.
	
	* Fixed a couple of minor POD formatting errors.
	
1.10 - Thu Jun  8 06:51:25 2006
	* Distro cleanup (removed sf_login.t test), fixed doc bug about
	PAUSE, cleansed POD. No need to upgrade.

1.08 - Sat May 20 03:04:01 2006
	* Added MANIFEST checking feature. If `make manifest` has to change
	MANIFEST, check_manifest() dies. You'll have to check in MANIFEST
	before you can go on anyway, so you might as well stop and take a
	look at it (and maybe update MANIFEST.SKIP too). I needed this 
	because although I had a lot of pod_coverage tests lying about, 
	they never made it into the dists because I never added them to
	MANIFEST. Stoopid me.
	* This updates the release(1) script too. Make sure you want this
	feature before you overwrite your old release(1).

1.07 - Wed May 17 21:06:39 2006
	* Minor updates to the distro - no code changes. No need to upgrade

0.25 - Fri Dec 17 15:53:15 CST 2004
	* No changes to functionality.
	* Released only because I (Andy) deleted the latest version,
	  because I thought brian had newer ones, when he didn't.
	  This meant that the indexer got all confused.  I hope this
	  stops people from now wondering "Where did Module::Release go?"
	* Modernized t/pod.t
	* Made tests in t/load.t more inclusive
	* Updated the Makefile.PL

0.24 - Sat Nov 29 09:29:36 2003
	* fixed a problem in Module::Release's check_cvs routine where
	it was not checking each line of output

0.23 - Sun Jun 22 23:40:00 2003 (Andy Lester)
	* All output now goes to the screen if --debug is on.  Thanks to
	Ken Williams.  "High school students."

0.23
	* release will show the user all STDOUT of the subprocesses
	  whenever debug is on.  This will prevent the machine from waiting
	  for a response from you because you don't know it's there.

0.22 - Wed Apr  9 16:05:08 2003 (Andy Lester)
	* release will not run if there are any switch errors

	* Added many error checks in ftp_upload.
    
	* Added some docs on how to find sf_group_id and sf_release_id.

	* Removed a "last CVS" in cvs() method.  It would have
  	  blown up if someone didn't have a CVS directory.

	* Renamed perl() method to build_makefile()

	* Renamed cvs() to check_cvs()

	* Added basic POD docs for the functions

0.21 - Sat Mar 29 16:19:00 2003 (Andy Lester)

Makefile.PL
# $Id: Makefile.PL 2668 2008-08-15 14:42:24Z comdog $
use ExtUtils::MakeMaker;

require 5.008;

eval "use Test::Manifest 1.14";

WriteMakefile(
	'NAME'	       => 'Module::Release',
	'AUTHOR'       => 'brian d foy <bdfoy@cpan.org>',
	'ABSTRACT'     => 'Automate software releases',
	'VERSION_FROM' => "lib/Release.pm",
	'LICENSE'      => 'perl',

	'EXE_FILES'    =>  [ 'script/release' ],

	'PREREQ_PM'    => 
		{
		'ConfigReader::Simple' => 0,
		'Net::FTP'             => 0,
		'HTTP::Message'        => 0,
		'IO::Null'             => 0,
		'Crypt::SSLeay'        => 0,
		'Test::Output'         => 0,
		},

	'PM'           => {
		'lib/Release.pm' => '$(INST_LIBDIR)/Release.pm',
		'lib/SVN.pm'     => '$(INST_LIBDIR)/Release/SVN.pm',
		'lib/FTP.pm'     => '$(INST_LIBDIR)/Release/FTP.pm',
		'lib/PAUSE.pm'   => '$(INST_LIBDIR)/Release/PAUSE.pm',
		'lib/Kwalitee.pm'   => '$(INST_LIBDIR)/Release/Kwalitee.pm',
		#map { "lib/$_.pm" => "\$(INST_LIBDIR)/Release/$_.pm" } qw( SVN PAUSE FTP ),
		},

	'MAN1PODS'     => {
		'script/release' => '$(INST_MAN1DIR)/release.1',
		},

	'MAN3PODS'     => {
		'lib/Release.pm' => '$(INST_MAN3DIR)/Module::Release.3',
		'lib/SVN.pm'     => '$(INST_MAN3DIR)/Module::Release::SVN.3',
		},

	clean => { FILES => '*.bak release-* Module-*' },
	);
